vunit/sim_if/xcelium.py

```diff
@@ -0,0 +1,694 @@
+
+        source_files_to_skip = set()
+
+        max_library_name = 0
+        max_source_file_name = 0
+        if source_files:
+            max_library_name = max(len(source_file.library.name) for source_file in source_files)
+            max_source_file_name = max(len(simplify_path(source_file.name)) for source_file in source_files)
+
+        source_files_by_library: Dict[Library, List[SourceFile]] = {}
+        for source_file in source_files:
+            if source_file.library in source_files_by_library:
+                source_files_by_library[source_file.library].append(source_file)
+            else:
+                source_files_by_library[source_file.library] = [source_file]
+        # import pprint
+        # pprint.pprint(source_files_by_library)
+
+        printer.write("Compiling all source files")
+        sys.stdout.flush()
+        if self._compile_all_source_files(source_files_by_library, printer):
+            printer.write("All source files compiled!\n")
+            printer.write("Compile passed\n", fg="gi")
+        else:
+            printer.write("One or more source files failed to compile.\n")
+            printer.write("Compile failed\n", fg="ri")
+            raise CompileError
+
+
+    @staticmethod
+    def _vhdl_std_opt(vhdl_standard):
+        """
+        Convert standard to format of xrun command line flag
+        """
+        if vhdl_standard == VHDL.STD_2002:
+            return "-v200x -extv200x"
+
+        if vhdl_standard == VHDL.STD_2008:
+            return "-v200x -extv200x -inc_v200x_pkg"
+
+        if vhdl_standard == VHDL.STD_1993:
+            return "-v93"
+
+        raise ValueError("Invalid VHDL standard %s" % vhdl_standard)
+
+    def _compile_all_files_in_library_subcommand(self, library, source_files):
+        """
+        Return a command to compile all source files in a library
+        """
+        args = []
+        args += ["-makelib %s" % library.directory]
+        search_list = []
+
+        for source_file in source_files:
+            unique_args = []
+            if source_file.is_vhdl:
+                unique_args += ["%s" % self._vhdl_std_opt(source_file.get_vhdl_standard())]
+                unique_args += source_file.get_compile_option("xcelium.xrun_vhdl_flags", include_global=False)
+
+            if source_file.is_any_verilog:
+                unique_args += source_file.get_compile_option("xcelium.xrun_verilog_flags", include_global=False)
+                for include_dir in source_file.include_dirs:
+                    if include_dir not in search_list:
+                        search_list.append(include_dir)
+                for key, value in source_file.defines.items():
+                    unique_args += ["-define %s=%s" % (key, value.replace('"', '\\"'))]
+
+            args += [source_file.name]
+            if len(unique_args) > 0:
+                # args += ["-filemap %s" % source_file.name]
```

Comment:
It'd be nice to remove commented code if it's not needed anymore

Reply:
We left this commented code in because we didn't have time to setup a test for it but we'd ideally like to support per-source file arguments as is supported by Xcelium.

Reply:
I see. In that case, I'd still remove it, but open a new issue with some of this code and where it should go unless you plan to add support.